src/lib.rs

```diff
@@ -220,6 +220,33 @@ impl Colorable for DirEntry {
 
 const LS_COLORS_DEFAULT: &str = "rs=0:lc=\x1b[:rc=m:cl=\x1b[K:ex=01;32:sg=30;43:su=37;41:di=01;34:st=37;44:ow=34;42:tw=30;42:ln=01;36:bd=01;33:cd=01;33:do=01;35:pi=33:so=01;35:";
 
+#[derive(Default, Debug, Clone)]
+struct SuffixMapping {
+    mappings: HashMap<FileNameSuffix, Option<Style>>,
+    case_sensitive: HashSet<FileNameSuffix>,
+}
+
+impl SuffixMapping {
+    fn push(&mut self, suffix: FileNameSuffix, style: Option<Style>) {
+        let normalized_suffix = suffix.to_ascii_lowercase();
+        // decides whether new mapping and it's varaints should be treated as case-sensitive
+        let is_case_sensitive = self.mappings.iter().any(|(m_suffix, m_style)| {
```

Comment:
It would be nice to avoid $`O(n^2)`$ complexity here, since some people have pretty large `$LS_COLORS`

Reply:
okay let me try 😀

Reply:
I managed to reduce the complexity, now when adding a single entry it doesn't compare with each entry just with its variants, which I guess in real world situations should be tiny compared to `$LS_COLORS`.